langchain/llms/huggingface_pipeline.py

```diff
@@ -0,0 +1,103 @@
+"""Wrapper around HuggingFace APIs."""
+from typing import Any, Dict, List, Mapping, Optional
+
+from pydantic import BaseModel, Extra, root_validator
+
+from langchain.llms.base import LLM
+from langchain.llms.utils import enforce_stop_tokens
+
+DEFAULT_MODEL_ID = "gpt2"
+VALID_TASKS = ("text2text-generation", "text-generation")
+
+
+class HuggingFacePipeline(LLM, BaseModel):
+    """Wrapper around HuggingFace Pipeline API.
+
+    To use, you should have the ``transformers`` python package installed.
+
+    Only supports `text-generation` and `text2text-generation` for now.
+
+    Example:
+        .. code-block:: python
+
+            from langchain import HuggingFacePipeline
+            hf = HuggingFacePipeline(model_id="gpt2")
+    """
+
+    pipeline: Any  #: :meta private:
+    model_id: str = DEFAULT_MODEL_ID
+    """Model name to use."""
+    task: Optional[str] = None
+    """Task to call the model with. Should be a task that returns `generated_text`."""
+    model_kwargs: Optional[dict] = None
+    """Key word arguments to pass to the model."""
+
+    class Config:
+        """Configuration for this pydantic object."""
+
+        extra = Extra.forbid
+
+    @root_validator()
+    def validate_environment(cls, values: Dict) -> Dict:
+        """Validate python package exists in environment."""
+        try:
+            from transformers import AutoModelForCausalLM, AutoTokenizer
+            from transformers import pipeline as hf_pipeline
+
+            model_id = values["model_id"]
+            tokenizer = AutoTokenizer.from_pretrained(model_id)
+            model = AutoModelForCausalLM.from_pretrained(model_id)
+
+            pipeline = hf_pipeline(
+                values["task"],
+                model=model,
+                tokenizer=tokenizer,
+                model_kwargs=values["model_kwargs"],
+            )
+            if pipeline.task not in VALID_TASKS:
+                raise ValueError(
+                    f"Got invalid task {pipeline.task}, "
+                    f"currently only {VALID_TASKS} are supported"
+                )
+            values["pipeline"] = pipeline
+
+        except ImportError:
+            raise ValueError(
+                "Could not import transformers python package. "
+                "Please it install it with `pip install transformers`."
+            )
+
+        return values
+
+    @property
+    def _identifying_params(self) -> Mapping[str, Any]:
+        """Get the identifying parameters."""
+        _model_kwargs = self.model_kwargs or {}
+        return {
+            **{"model_id": self.model_id, "task": self.task},
+            **{"model_kwargs": _model_kwargs},
+        }
+
+    @property
+    def _llm_type(self) -> str:
+        return "huggingface_local"
+
+    def _call(self, prompt: str, stop: Optional[List[str]] = None) -> str:
+        response = self.pipeline(text_inputs=prompt)
+        if "error" in response:
```

Comment:
can this still happen with the local pipeline? probably not right? not sure we need this check

Reply:
done